x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_panel.tsx

```diff
@@ -202,14 +232,19 @@ export function SuggestionPanel({
           )
         : undefined;
 
-    return { suggestions: newSuggestions, currentStateExpression: newStateExpression };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This was kept on purpose - otherwise the suggestions are recalculated too often because the frame object gets recreated on each render. This hook should not depend on the frame object. Maybe this is causing the issue with `toExpression` I mentioned in the other comment.